Bumping `mailparse` crate

Following https://github.com/staktrace/mailparse/issues/108#issuecomment-1338818182, is it possible to bump your version here also?

Because I am using both `mailparse` and `maildir` in my project, and it creates some conflicts:

```
error[E0308]: mismatched types
   --> src/domain/envelope/maildir/envelope.rs:46:61
    |
46  |                     let addrs = mailparse::addrparse_header(h)
    |                                 --------------------------- ^ expected struct `MailHeader`, found struct `mailparse::MailHeader`
    |                                 |
    |                                 arguments to this function are incorrect
    |
    = note: expected reference `&MailHeader<'_>`
               found reference `&mailparse::MailHeader<'_>`
    = note: perhaps two different versions of crate `mailparse` are being used?
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -17,9 +17,9 @@ exclude = [".gitignore", ".github/**"]
 maintenance = { status = "passively-maintained" }
 
 [dependencies]
-mailparse = "0.13"
-gethostname = "0.2.1"
-memmap2 = { version = "0.5.7", optional = true }
+mailparse = "0.14"
+gethostname = "0.4.1"
+memmap2 = { version = "0.5.8", optional = true }
 
 [features]
 mmap = ["memmap2"]

